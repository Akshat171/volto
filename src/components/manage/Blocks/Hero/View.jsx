/**
 * View hero block.
 * @module components/manage/Blocks/Hero/View
 */

import React from 'react';
import { withBlockExtensions } from '@plone/volto/helpers';
import { HeroBody } from '@plone/volto/components';
import cx from 'classnames';

export const View = (props) => {
  const { data } = props;
  return (
    <div
      className={cx(
        'block hero align',
        {
          // we keep the default alignment of the original hero block
          left: !Boolean(data.align),
        },
        data.align,
      )}
    >
      <HeroBody {...props}></HeroBody>
    </div>
  );
};

export default withBlockExtensions(View);
