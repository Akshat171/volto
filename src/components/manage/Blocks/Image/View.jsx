/**
 * View image block.
 * @module components/manage/Blocks/Image/View
 */

import React from 'react';
import PropTypes from 'prop-types';
import { UniversalLink } from '@plone/volto/components';
import cx from 'classnames';
import Image from '@plone/volto/components/theme/Image/Image';

/**
 * View image block class.
 * @class View
 * @extends Component
 */
const View = ({ data, detached }) => (
  <p
    className={cx(
      'block image align',
      {
        center: !Boolean(data.align),
        detached,
      },
      data.align,
    )}
  >
    {data.url && (
      <>
        {(() => {
          const image = (
            <Image
              image={data.url}
              alt={data.alt}
              className={cx({
                'full-width': data.align === 'full',
                large: data.size === 'l',
                medium: data.size === 'm',
                small: data.size === 's',
              })}
            />
          );
          if (data.href) {
            return (
              <UniversalLink
                href={data.href}
                openLinkInNewTab={data.openLinkInNewTab}
              >
                {image}
              </UniversalLink>
            );
          } else {
            return image;
          }
        })()}
      </>
    )}
  </p>
);

/**
 * Property types.
 * @property {Object} propTypes Property types.
 * @static
 */
View.propTypes = {
  data: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default View;
