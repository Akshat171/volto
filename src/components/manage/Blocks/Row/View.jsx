import React from 'react';
import PropTypes from 'prop-types';
import { Grid } from 'semantic-ui-react';
import cx from 'classnames';
import { RenderBlocks } from '@plone/volto/components';
import { withBlockExtensions } from '@plone/volto/helpers';

const RowBlockView = (props) => {
  const { data } = props;
  const metadata = props.metadata || props.properties;
  const columns = data.data.blocks_layout.items;

  return (
    <div
      className={cx('block __grid', {
        [data['@type']]: data['@type'] !== '__grid',
        centered: data.align === 'center' || data.align === undefined,
        'space-between': data.align === 'space-between',
        'centered-text': data.centeredText,
        one: columns?.length === 1,
        two: columns?.length === 2,
        three: columns?.length === 3,
        four: columns?.length === 4,
      })}
    >
      {data.headline && <h2 className="headline">{data.headline}</h2>}

      <Grid stackable stretched columns={columns.length}>
        <RenderBlocks
          {...props}
          blockWrapperTag={Grid.Column}
          metadata={metadata}
          content={data.data}
        />
      </Grid>
    </div>
  );
};

/**
 * Property types.
 * @property {Object} propTypes Property types.
 * @static
 */
RowBlockView.propTypes = {
  data: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default withBlockExtensions(RowBlockView);
