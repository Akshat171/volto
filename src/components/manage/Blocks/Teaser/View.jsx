import React from 'react';
import TeaserBody from './Body';
import { withBlockExtensions } from '@plone/volto/helpers';

const TeaserBlockView = (props) => {
  return <TeaserBody {...props} />;
};

export default withBlockExtensions(TeaserBlockView);
