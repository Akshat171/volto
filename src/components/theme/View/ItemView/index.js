import NewsItemItemView from './NewsItemItemView';
import EventItemView from './EventItemView';
import DefaultItemView from './DefaultItemView';

export { NewsItemItemView, DefaultItemView, EventItemView };
