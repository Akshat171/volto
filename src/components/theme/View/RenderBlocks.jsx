import React from 'react';
import { getBaseUrl, applyBlockDefaults } from '@plone/volto/helpers';
import { defineMessages, injectIntl } from 'react-intl';
import { map } from 'lodash';
import { MaybeWrap } from '@plone/volto/components';
import {
  getBlocksFieldname,
  getBlocksLayoutFieldname,
  hasBlocksData,
} from '@plone/volto/helpers';
import config from '@plone/volto/registry';

const messages = defineMessages({
  unknownBlock: {
    id: 'Unknown Block',
    defaultMessage: 'Unknown Block {block}',
  },
});

const RenderBlocks = (props) => {
  const { path, intl, content, metadata, as, blockWrapperTag } = props;
  const blocksFieldname = getBlocksFieldname(content);
  const blocksLayoutFieldname = getBlocksLayoutFieldname(content);
  const blocksConfig = props.blocksConfig || config.blocks.blocksConfig;

  return hasBlocksData(content) ? (
    <MaybeWrap condition={as} as={as}>
      {map(content[blocksLayoutFieldname].items, (block) => {
        const Block =
          blocksConfig[content[blocksFieldname]?.[block]?.['@type']]?.view;

        const blockData = applyBlockDefaults({
          data: content[blocksFieldname][block],
          intl,
          metadata,
          properties: content,
        });

        return Block ? (
          <MaybeWrap condition={blockWrapperTag} as={blockWrapperTag}>
            <Block
              key={block}
              id={block}
              metadata={metadata}
              properties={content}
              data={blockData}
              path={getBaseUrl(path || '')}
              blocksConfig={blocksConfig}
            />
          </MaybeWrap>
        ) : (
          <div key={block}>
            {intl.formatMessage(messages.unknownBlock, {
              block: content[blocksFieldname]?.[block]?.['@type'],
            })}
          </div>
        );
      })}
    </MaybeWrap>
  ) : (
    ''
  );
};

export default injectIntl(RenderBlocks);
