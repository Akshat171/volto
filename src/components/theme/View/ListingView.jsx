/**
 * Document view component.
 * @module components/theme/View/ListingView
 */

import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { Segment, Container } from 'semantic-ui-react';
import Image from '@plone/volto/components/theme/Image/Image';

/**
 * List view component class.
 * @function ListingView
 * @params {object} content Content object.
 * @returns {string} Markup of the component.
 */
const ListingView = ({ content }) => (
  <Container id="page-home">
    <section id="content-core">
      {content.items.map((item) => (
        <Segment key={item.url} className="listing-item">
          <Container>
            <h2>
              <Link to={item.url} title={item['@type']}>
                {item.title}
              </Link>
            </h2>
            {item.description && <p>{item.description}</p>}
          </Container>
          {item.image_field && (
            <Image
              image={item['@id']}
              size="thumb"
              floated="right"
              imageField={item.image_field}
            />
          )}
        </Segment>
      ))}
    </section>
  </Container>
);

/**
 * Property types.
 * @property {Object} propTypes Property types.
 * @static
 */
ListingView.propTypes = {
  content: PropTypes.shape({
    title: PropTypes.string,
    description: PropTypes.string,
    items: PropTypes.arrayOf(
      PropTypes.shape({
        '@id': PropTypes.string,
        '@type': PropTypes.string,
        description: PropTypes.string,
        review_state: PropTypes.string,
        title: PropTypes.string,
        url: PropTypes.string,
      }),
    ),
  }).isRequired,
};

export default ListingView;
