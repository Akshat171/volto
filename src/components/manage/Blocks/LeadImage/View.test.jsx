import React from 'react';
import renderer from 'react-test-renderer';
import View from './View';

test('renders a view image component', () => {
  const component = renderer.create(
    <View
      data={{}}
      properties={{
        image: {
          scales: {
            preview: {
              download: 'image.png',
              width: 400,
              height: 400,
            },
          },
        },
      }}
    />,
  );
  const json = component.toJSON();
  expect(json).toMatchSnapshot();
});
