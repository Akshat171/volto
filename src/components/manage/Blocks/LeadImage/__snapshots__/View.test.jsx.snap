// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a view image component 1`] = `
<p
  className="block image align center"
>
  <picture
    className="volto-image responsive"
  >
    <img
      alt=""
      className=""
      role="img"
      src="http://localhost:8080/Plone/test-images/@@images/image/listing"
      style={null}
    />
  </picture>
  <noscript
    dangerouslySetInnerHTML={
      Object {
        "__html": "
              <img
                src=\\"http://localhost:8080/Plone/test-images/@@images/image/listing\\"
                srcset=\\"/test-images/@@images/image/listing 16w, /test-images/@@images/image/icon 32w, /test-images/@@images/image/tile 64w, /test-images/@@images/image/thumb 128w, /test-images/@@images/image/mini 200w, /test-images/@@images/image/preview 400w, /test-images/@@images/image/large 768w\\"
                alt=\\"\\"
                class=\\"\\"
                role=\\"img\\"
                
                                         
                loading=\\"lazy\\"               
            ",
      }
    }
  />
</p>
`;
